src/components/utils/onfidoApi.js

```diff
@@ -1,8 +1,14 @@
 import { performHttpReq } from '../utils/http'
 import forEach from 'object-loops/for-each'
 
-const formatError = ({response, status}, onError) =>
-  onError({status, response: JSON.parse(response)})
+const formatError = ({response, status}, onError) => {
+  try {
+    onError({status, response: JSON.parse(response)})
+  }
+  catch {
+    onError({status, response: {}})
```

Comment:
how does the upstream (e.g. popups) handle an empty error object?

Reply:
For any error which is not 401 or 422, we ignore the response object and return a generic "Connection Lost" error

Reply:
so I'm guessing we would never get a 401 or 422 that doesn't have a json error object, right?

Reply:
Not from the API I would hope. But if for whatever reason the API returns a non json response we would not be able to map the error, therefore we would return a generic "Connection Lost" error back to the user.